### PR TITLE
Add org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

### PR DESCRIPTION
**Pull request by Bleuzen**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Bleuzen** [comment]
This is how far I got packaging ChowTapeModel. However, it is not ready for merge (may even have to make another PR). Will add more info soon.

For now, it doesn't even build:

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by Bleuzen** [comment]
From build log:
```
-- Checking for modules 'webkit2gtk-4.0;gtk+-x11-3.0'
--   Package 'webkit2gtk-4.0', required by 'virtual:world', not found
```
May have to disable this gtk stuff from JUCE if that is possible.

**Comment by Bleuzen** [comment]
Hm from [this comment](https://github.com/jatinchowdhury18/AnalogTapeModel/issues/7#issuecomment-657914365):

> Alright, I've now disabled JUCE_WEB_BROWSER in the submodule fork of the Projucer, so the webkit2gtk dependency can be removed altogether.

It should already build without webkit2gtk, so maybe this is irrelevant.

At the end of the log there is another problem, it doesn't find LV2:

```
In file included from /run/build/ChowTapeModel/Plugin/modules/JUCE/modules/juce_audio_plugin_client/juce_audio_plugin_client_LV2.cpp:27:
/run/build/ChowTapeModel/Plugin/modules/JUCE/modules/juce_audio_plugin_client/LV2/juce_LV2_Wrapper.cpp:54:10: fatal error: lv2/lv2plug.in/ns/lv2core/lv2.h: No such file or directory
   54 | #include <lv2/lv2plug.in/ns/lv2core/lv2.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [CMakeFiles/CHOWTapeModel_LV2.dir/build.make:90: CMakeFiles/CHOWTapeModel_LV2.dir/modules/JUCE/modules/juce_audio_plugin_client/juce_audio_plugin_client_LV2.cpp.o] Error 1
make[3]: Leaving directory '/run/build/ChowTapeModel/Plugin/build'
make[2]: *** [CMakeFiles/Makefile2:264: CMakeFiles/CHOWTapeModel_LV2.dir/all] Error 2
make[2]: Leaving directory '/run/build/ChowTapeModel/Plugin/build'
make[1]: *** [CMakeFiles/Makefile2:271: CMakeFiles/CHOWTapeModel_LV2.dir/rule] Error 2
make[1]: Leaving directory '/run/build/ChowTapeModel/Plugin/build'
make: *** [Makefile:170: CHOWTapeModel_LV2] Error 2
make: Leaving directory '/run/build/ChowTapeModel/Plugin/build'
]2;flatpak-builder: Cleanup ChowTapeModelError: module ChowTapeModel: Child process exited with code 2
```

Don't know why, lv2 should be there:
https://github.com/flathub/flathub/pull/2681/files#diff-73fe21098234989ddc5d816a1d6fe70056e1fa87403cc77f9fb75f182d6dd29fR25

**Comment by barthalion** [comment]
Any idea @hfiguiere?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
See my changes to have this build properly.

The plugins show up in Ardour and the UI show up. I have tested whether they actually do something yet.

https://github.com/hfiguiere/flathub/commit/71b74994017f80ad44e3c05bdcac75cdb39be41c

**Comment by hfiguiere** [comment]
Just a few details:
- the build still build a "standalone" version of the effect. Due to limitation with flatpak this would need to be a separate package. It's deleted anyway
- I use `cmake-ninja` with the right options
- AR, NM and RANLIB are needed for lto otherwise the build fail. Not sure why CMake doesn't pick that up
- `cxxflags` is set so that lv2 can be found as the build system doesn't use pkgconfig like it should
- Plugins are never installed so we copy them manually
- Like other plugins, we need to strip them until such a day were flatpak-builder build the debug package
- Fixed some install path

File: org.freedesktop.LinuxAudio.Plugins.ChowTapeModel.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.ChowTapeModel</id>
+  <extends>org.freedesktop.LinuxAudio.BaseExtension</extends>
+  <name>ChowTapeModel</name>
+  <summary>Physical modelling signal processing for analog tape recording</summary>
+  <url type="homepage">https://chowdsp.com/</url>
+  <project_license>GPL-3.0</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
+  <releases>
+    <release date="2021-12-05" version="2.10.0" />
```
[the remark this answers is not in the document]
Comment (Bleuzen) [reply org.freedesktop.LinuxAudio.Plugins.ChowTapeModel.metainfo.xml:11]: added the "is-main-source" flag now, this should do it

File: org.freedesktop.LinuxAudio.Plugins.ChowTapeModel.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="addon">
+  <id>org.freedesktop.LinuxAudio.Plugins.ChowTapeModel</id>
+  <extends>org.freedesktop.LinuxAudio.BaseExtension</extends>
+  <name>ChowTapeModel</name>
+  <summary>Physical modelling signal processing for analog tape recording</summary>
```
[the remark this answers is not in the document]
Comment (Bleuzen) [reply org.freedesktop.LinuxAudio.Plugins.ChowTapeModel.metainfo.xml]: I applied your commit which includes this, thanks

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by hfiguiere** [comment]
ah x86_64 portability.....

**Comment by hfiguiere** [comment]
it uses xsimd though, so it's probably some tweaking needed.

**Comment by hfiguiere** [comment]
`vvtanh` is missing and from what I gather it's because they assume aarch64 == Apple.

I think until there is a patch restricting arch might be the way to get this in.

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by hfiguiere** [comment]
I have two patches to build on aarch64 (untested at runtime). I'll submit a PR later for it.

**Comment by hfiguiere** [comment]
Patch one https://github.com/jatinchowdhury18/AnalogTapeModel/pull/237
Patch two https://github.com/Chowdhury-DSP/chowdsp_utils/pull/98

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by Bleuzen** [comment]
@hfiguiere 
> the build still build a "standalone" version of the effect

> I use `cmake-ninja` with the right options

Is there no way to set build targets with that buildsystem? Like it was done before with cmake+make:
https://github.com/Bleuzen/flathub/blob/870db63d5c53aa2072fe65558008b9eaba4385dd/org.freedesktop.LinuxAudio.Plugins.ChowTapeModel.json#L32

**Comment by Bleuzen** [comment]
@hfiguiere 
> the build still build a "standalone" version of the effect

> I use cmake-ninja with the right options

Is there no option to set the build targets with `cmake-ninja`?

If not, thinking about to switch build systems... not sure which one is the best yet. Could use cmake, but afaik from developers examples, it can only build one target at a time, so would need to be executed two times, which... may be not possible in flatpak? So may have to switch back to simple build system.

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by Bleuzen** [comment]
@nedrichards  @barthalion  maybe one of you can help...
I try to set build targets (only want to build `CHOWTapeModel_VST3` and `CHOWTapeModel_LV2`). The recommended cmake commands would look like this:
```
cmake --build build --target CHOWTapeModel_VST3 ...
cmake --build build --target CHOWTapeModel_LV2 ...
```

But I don't know how to set that target option in a flatpak manifest, especially with the current `cmake-ninja` buildsystem. Is there a way?

**Comment by hfiguiere** [comment]
`make-args` as documented in `man flatpak-manifest`

But when I did that it ddn't change anything.

**Comment by Bleuzen** [comment]
From docs:
> An array of extra arguments that will be passed to make

Thought it is only for make, not ninja.

**Comment by Bleuzen** [comment]
I also gave it a try with `make-args` with both: `cmake` and `cmake-ninja` buildsystems. All targets got built, even tho I added the desired targets only as make-args. However this works when building outside of flatpak. Using make + a target as argument builds only that target.
So seems something wrong with flatpak.
Is there a limitation, something like "everything in make-args has to start with a dash or will be ignored" or is this a bug in flatpak?

**Comment by Bleuzen** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by nedrichards** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.LinuxAudio.Plugins.ChowTapeModel

**Comment by hfiguiere** [comment]
this also needed a branch

**Pull request closed without merging**